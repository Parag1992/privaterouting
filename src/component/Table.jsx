import React from "react";

const Table = (props) => {

    return(
        <React.Fragment>
            <div className="tblCls"> I am table Component </div><br/>
            <div className="prpCls">{props.myName}</div>
        </React.Fragment>
    )
}
export default Table;