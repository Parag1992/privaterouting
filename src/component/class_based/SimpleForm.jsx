import React from "react";
import Table from "../Table";
import { Navigate } from "react-router-dom";
import Header from "../Header";
class SimpleForm extends React.Component {

    constructor(props)
    {
        super(props);
        this.state={
            txtName:"Parag",
            dispTxtNm:""
        }
    }

    componentDidMount(){
      //  console.log('componentDidMount');
    }

    UpdateTxt = (e) =>{
        this.setState({txtName:e.target.value});
    }

    onBtnClicked =() =>{
        let tmpNm=this.state.txtName;
        this.setState({dispTxtNm:tmpNm});
        
        return 3;
    }

    asyncFunc = (a) =>{
        return new Promise((resolve,reject) => {
            //let a=10;
            if(a%2==0){
                resolve("I am resolved");
            }else{
                reject("I have rejected");
            }
        })

        
    }

    render(){
        return (
        <React.Fragment>
           
             <h1 className="h1Cls">My Name is : {this.state.dispTxtNm}</h1><br/>
             <input type={"text"} key={"txtNm"} onChange={this.UpdateTxt} />
             <button className="myBtnCls" value={"Submit"} onClick={this.onBtnClicked} >Submit</button><br/>
             {this.state.dispTxtNm == "abc" &&  <Navigate to="/todo" replace={true} />}
             <hr/>
             <Table myName="I am  props"/>
             
        </React.Fragment>

    )}
}
export default SimpleForm;