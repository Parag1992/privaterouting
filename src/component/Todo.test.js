import TodoApp from './Todo';
import { mount, shallow } from 'enzyme';
import React from "react";
import SimpleForm from './class_based/SimpleForm';
import Table from './Table';

describe("Todo App shallow testing ",()=>{

   test("function testing", ()=>{
      let wrapper = shallow(<TodoApp/>);
      wrapper.find('.input').simulate('change',{target:'abc'})
      wrapper.find('.btnCls').simulate('click');
      console.log(wrapper.debug());
      //expect(wrapper.find('.abc').text()).toBe('abc');
   })
   //  test('adds 1 + 2 to equal 3', () => {
   //     let wrapper=shallow(<TodoApp/>)
   //     //console.log(wrapper.debug());
   //     expect(wrapper.exists('.todoCls')).toEqual(true);
   //  });

   //  test('check useState', () => {
       
   //      const setState = jest.fn();
   //      const useStateSpy = jest.spyOn(React, "useState");
   //      useStateSpy.mockImplementation((initialState) => [initialState, setState]);
   //      let wrapper=shallow(<TodoApp/>)
   //      const newInputValue = "React is Awesome";
   //      wrapper
   //          .find(".input")
   //          .simulate("change", { target: { value: newInputValue } });
   //          console.log(setState);
   //      expect(setState).toHaveBeenCalledWith(newInputValue);
   //      //expect(setState).toEqual([]);
   //   });


     test('check class based comp state',()=>{
        let wrapper=shallow(<SimpleForm/>);
        expect(wrapper.state()).toEqual({txtName:"Parag",dispTxtNm:""});
     })

     test('check class based comp state',()=>{
      let wrapper=shallow(<SimpleForm/>);
      expect(wrapper.state().txtName).toEqual("Parag");
   })

   test('check comp props',()=>{
      let propValue="I am props";
      let wrapper=shallow(<Table myName={propValue}/>);
      expect(wrapper.find('.prpCls').text()).toEqual(propValue);
   })

   //   test('check html element is present',()=>{
   //    let wrapper=shallow(<SimpleForm/>);
   //    expect(wrapper.contains(<h1>My Name is : Parag</h1>)).toEqual(true);
   // })

   // test('check html element text message only',()=>{
   //    let wrapper=shallow(<SimpleForm/>);
   //    expect(wrapper.containsAllMatchingElements([<h1>My Name is : Parag</h1>])).toEqual(true);
   // })

   test('Simulate button to check text message ',()=>{
      let wrapper=shallow(<SimpleForm/>);
      wrapper.find('.myBtnCls').simulate('click');
      expect(wrapper.find('.h1Cls').text()).toEqual('My Name is : Parag');
   })

   test('dive to test inner comp ',()=>{
      let wrapper=shallow(<SimpleForm/>);
    
      expect(wrapper.find(Table).dive().exists('.tblCls')).toBe(true);
   })

   test('Snapshot testing ',()=>{
      let wrapper=shallow(<SimpleForm/>);
    
      expect(wrapper).toMatchSnapshot();
   })
})

describe.skip("Todo App Mount testing",()=>{

   beforeAll(()=>{
      console.log("Before All");
   })

   beforeEach(()=>{
      console.log("Before Each");
   })
   

   test('Mount testing', ()=>{
      let wrapper = mount(<SimpleForm/>);

      //console.log(wrapper.debug());
      expect(wrapper.exists('.tblCls')).toBe(true);
   })

   test('Testing using Instance ', ()=>{
      let wrapper = shallow(<SimpleForm/>);

      let testInstance = wrapper.instance().onBtnClicked();
      expect(testInstance).toBe(3);
   })

   test('Testing async func using Instance ', ()=>{
      let wrapper = shallow(<SimpleForm/>);

      return wrapper.instance().asyncFunc(4).then(data=>{
         expect(data).toBe("I am resolved");
      });
      
   })

   test('Testing using jest.spyOn lifecyle method ', ()=>{
      jest.spyOn(SimpleForm.prototype,'componentDidMount');
      let wrapper = shallow(<SimpleForm/>);
      expect(SimpleForm.prototype.componentDidMount).toHaveBeenCalled();
   })

   test.skip('Testing using jest.spyOn user created method ', ()=>{      
      let wrapper = shallow(<SimpleForm/>);
      let instance=wrapper.instance();
      jest.spyOn(instance,'func');
      wrapper.find('button').simulate('click');
      expect(instance.func).toHaveBeenCalled();
   })
})