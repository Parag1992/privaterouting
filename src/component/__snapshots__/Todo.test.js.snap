// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo App shallow testing  Snapshot testing  1`] = `
<Fragment>
  <h1
    className="h1Cls"
  >
    My Name is : 
  </h1>
  <br />
  <input
    key="txtNm"
    onChange={[Function]}
    type="text"
  />
  <button
    className="myBtnCls"
    onClick={[Function]}
    value="Submit"
  >
    Submit
  </button>
  <br />
  <hr />
  <Table
    myName="I am  props"
  />
</Fragment>
`;
